packages/plugin-ext/src/hosted/node/hosted-plugins-manager.ts

```diff
@@ -0,0 +1,149 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from "inversify";
+import * as cp from "child_process";
+import * as fs from "fs";
+import { sep as PATH_SEPARATOR } from "path";
+import { HostedPluginSupport } from "./hosted-plugin";
+import { LogType } from "../../common/types";
+
+export const HostedPluginsManager = Symbol('HostedPluginsManager');
+
+export interface HostedPluginsManager {
+
+    /**
+     * Runs watcher script to recomple plugin on any changes along given path.
+     *
+     * @param uri uri to plugin root folder.
+     */
+    runWatchCompilation(uri: string): Promise<void>;
+
+    /**
+     * Stops watcher script.
+     *
+     * @param uri uri to plugin root folder.
+     */
+    stopWatchCompilation(uri: string): Promise<void>;
+
+    /**
+     * Chacks if watcher script to recomple plugin is running.
+     *
+     * @param uri uri to plugin root folder.
+     */
+    isWatchCompilationRunning(uri: string): Promise<boolean>;
```

Comment:
What do you think about renaming above methods to `startWatching`, `stopWatching` and `isWatching` ??

Reply:
I think that we should keep it as is because `watch` could mean many things.